src/spikeinterface/core/binaryfolder.py

```diff
@@ -50,6 +50,9 @@ def __init__(self, folder_path):
 
         self._kwargs = dict(folder_path=str(folder_path.absolute()))
         self._bin_kwargs = d["kwargs"]
+        if "num_channels" not in self._bin_kwargs:
```

Comment:
Should this be removed as well?

Reply:
The assertion under I mean.

Reply:
either `num_channels` or `num_chan` should be in the dict right?

Reply:
Shouldn't it be only `num_channels` now?

Reply:
yes, but people have been saving in binary format for a while now! So I think this is needed for back-compatibility!

Reply:
Then we should add a warning that this will not be supported in SI >= 0.100?

Reply:
Ah, fair enough, makes sense!

Reply:
In my view a deprecation warning makes sense when loading old files makes sense.

Reply:
I mean it's already implemented in the `WaveformExtractor` load

Reply:
What would a deprecation warning do here? Let's say one saved a recording object with 0.97 and stored the folder ina remote drive/cloud. I don't think we need to require them to overwrite the object with a new version, but we should still be able to load it! 

If one uses a newer version of SI and passed the `num_chan` argument, (s)he will get the warning anyways!

Reply:
> I don't think we need to require them to overwrite the object with a new version, but we should still be able to load it!

Isn't that what we do with the `WaveformExtractor`?

Reply:
Where?

Reply:
Ah I guess it was removed. In the past when the version changed there was a warning with the previously-loaded waveform extractors :)